services/bettersocial/post/blockAnonymousPost.js

```diff
@@ -41,7 +42,6 @@ const BetterSocialBlockAnonymousPost = async (token, selfUserId, postId, source,
   }
 
   const authorAnonymousUserId = post?.actor?.id || null;
-
   console.log(`${selfAnonymousUserId?.user_id} vs ${authorAnonymousUserId}`);
   if (selfAnonymousUserId?.user_id === authorAnonymousUserId)
     return {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The comparison between `selfAnonymousUserId?.user_id` and `authorAnonymousUserId` has been added. Please make sure that both variables are of the same type (string, number, etc.) to avoid unexpected behavior due to JavaScript's loose equality checking (`==`). If they are not of the same type, consider converting them to a common type before comparison or use strict equality (`===`) if they are already of the same type.
```diff
+   const authorAnonymousUserId = post?.actor?.id || null;
+   console.log(`${selfAnonymousUserId?.user_id} vs ${authorAnonymousUserId}`);
+   if (selfAnonymousUserId?.user_id === authorAnonymousUserId)
+     return {
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->